fedimint-client-rpc/src/lib.rs

```diff
@@ -556,4 +563,38 @@ impl RpcGlobalState {
             Ok(None)
         }
     }
+
+    // Checks if the federation has a backup available
+    async fn check_backup_exists(&self, invite_code: String) -> anyhow::Result<bool> {
+        // Parse invite code and get federation id
+        let invite_code = InviteCode::from_str(&invite_code)?;
+        let federation_id = invite_code.federation_id();
+        let mnemonic = match self.get_mnemonic_from_db().await? {
+            Some(mnemonic) => mnemonic,
+            None => return Ok(false),
+        };
+
+        let federation_secret = self.derive_federation_secret(&mnemonic, &federation_id);
+        let client_config = fedimint_api_client::api::net::Connector::default()
+            .download_from_invite_code(&invite_code)
+            .await?;
+        let decoders = fedimint_core::module::registry::ModuleDecoderRegistry::default();
+
+        let api = fedimint_api_client::api::DynGlobalApi::from_endpoints(
+            client_config
+                .global
+                .api_endpoints
+                .iter()
+                .map(|(peer_id, peer_url)| (*peer_id, peer_url.url.clone())),
+            &invite_code.api_secret(),
+        )
+        .await?;
+        let backup = fedimint_client::Client::download_backup_from_federation_static(
+            &api,
+            &federation_secret,
+            &decoders,
+        )
+        .await?;
+        Ok(backup.is_some())
+    }
```

Comment:
Shouldn't this be implemented on `ClientPreview` and then used here? Or maybe a static fn in the core client that is used by both the `ClientPreview` impl and this one? Otherwise someone will eventually want the same functionality on the core client and we'll duplicate the code, not remembering it already exists in the RPC crate.